src/storage-controller/src/lib.rs

```diff
@@ -1129,7 +1126,85 @@ where
                 .expect("poisoned")
                 .insert(id, statistics::StatsInitState(BTreeMap::new()));
 
-            client.send(StorageCommand::CreateSinks(vec![cmd]));
+            client.send(StorageCommand::RunSinks(vec![cmd]));
+        }
+        Ok(())
+    }
+
+    /// Create the sinks described by the `ExportDescription`.
+    async fn update_export_connection(
+        &mut self,
+        exports: BTreeMap<GlobalId, StorageSinkConnection>,
+    ) -> Result<(), StorageError> {
+        for (id, connection) in exports {
+            let export = self.export_mut(id).expect("export exists");
+            let current_sink = export.description.sink.clone();
+            export.description.sink.connection = connection;
+
+            // Ensure compatibility
+            current_sink.alter_compatible(id, &export.description.sink)?;
+
+            // Get export data
+            let value = MetadataExportFetcher::get_stash_collection()
+                .peek_key_one(&mut self.stash, id.into_proto())
+                .await?
+                .expect("known to exist");
+
+            let mut durable_export_data = DurableExportMetadata::from_proto(value)
+                .map_err(|e| StorageError::IOError(e.into()))?;
+
+            // Get updated export state
+            let export = self.export(id).expect("export exists").clone();
```

Comment:
for clarity, id explicitly `drop(export)` to make it clear the shadowing drops the mutable access

Reply:
Refactoring for the comment below this one made that unnecessary––however, dropping a reference doesn't do anything so seems p odd.